airbyte-integrations/connectors/destination-mysql/src/test-integration/java/io/airbyte/integrations/destination/mysql/MySQLDestinationAcceptanceTest.java

```diff
@@ -260,58 +271,76 @@ protected void assertSameValue(final JsonNode expectedValue, final JsonNode actu
+    assertStringContains(status.getMessage(), "State code: 08S01;");
   }
 
   @Test
   public void testCheckIncorrectPortFailure() {
-    final JsonNode config = ((ObjectNode) getConfig()).put(JdbcUtils.PORT_KEY, "0000");
+    final JsonNode config = ((ObjectNode) getConfigForBareMetalConnection()).put(JdbcUtils.PORT_KEY, "0000");
     final MySQLDestination destination = new MySQLDestination();
     final AirbyteConnectionStatus status = destination.check(config);
     assertEquals(AirbyteConnectionStatus.Status.FAILED, status.getStatus());
-    assertTrue(status.getMessage().contains("State code: 08S01;"));
+    assertStringContains(status.getMessage(), "State code: 08S01;");
   }
 
   @Test
   public void testCheckIncorrectDataBaseFailure() {
-    final JsonNode config = ((ObjectNode) getConfig()).put(JdbcUtils.DATABASE_KEY, "wrongdatabase");
+    final JsonNode config = ((ObjectNode) getConfigForBareMetalConnection()).put(JdbcUtils.DATABASE_KEY, "wrongdatabase");
     final MySQLDestination destination = new MySQLDestination();
     final AirbyteConnectionStatus status = destination.check(config);
     assertEquals(AirbyteConnectionStatus.Status.FAILED, status.getStatus());
-    assertTrue(status.getMessage().contains("State code: 42000; Error code: 1049;"));
+    assertStringContains(status.getMessage(), "State code: 42000; Error code: 1049;");
   }
 
   @Test
   public void testUserHasNoPermissionToDataBase() {
     executeQuery("create user '" + USERNAME_WITHOUT_PERMISSION + "'@'%' IDENTIFIED BY '" + PASSWORD_WITHOUT_PERMISSION + "';\n");
-    final JsonNode config = ((ObjectNode) getConfig()).put(JdbcUtils.USERNAME_KEY, USERNAME_WITHOUT_PERMISSION);
+    final JsonNode config = ((ObjectNode) getConfigForBareMetalConnection()).put(JdbcUtils.USERNAME_KEY, USERNAME_WITHOUT_PERMISSION);
     ((ObjectNode) config).put(JdbcUtils.PASSWORD_KEY, PASSWORD_WITHOUT_PERMISSION);
     final MySQLDestination destination = new MySQLDestination();
     final AirbyteConnectionStatus status = destination.check(config);
     assertEquals(AirbyteConnectionStatus.Status.FAILED, status.getStatus());
-    assertTrue(status.getMessage().contains("State code: 42000; Error code: 1044;"));
+    assertStringContains(status.getMessage(), "State code: 42000; Error code: 1044;");
   }
 
+  private static void assertStringContains(final String str, final String target) {
+    assertTrue(str.contains(target), "Expected message to contain \"" + target + "\" but got " + str);
+  }
+
+  /**
+   * Legacy mysql normalization is broken, and uses the FLOAT type for numbers. This rounds off e.g.
+   * 12345.678 to 12345.7. We can fix this in DV2, but will not fix legacy normalization. As such,
+   * disabling the test case.
+   */
+  @Override
+  @Disabled
+  @ParameterizedTest
+  @ArgumentsSource(DataTypeTestArgumentProvider.class)
+  public void testDataTypeTestWithNormalization(final String messagesFilename,
```

Comment:
See Stephane's suggestion in #33507

Reply:
done in https://github.com/airbytehq/airbyte/pull/33468/commits/43c5d27eb6b47cc1accaf5bb1b4e0b59ac2cdca1